src/yunohost/app.py

```diff
@@ -65,66 +67,113 @@ def app_listlists():
     """
     List fetched lists
 
-
     """
-    list_list = []
-    try:
-        for filename in os.listdir(REPO_PATH):
-            if '.json' in filename:
-                list_list.append(filename[:len(filename)-5])
-    except OSError:
-        raise MoulinetteError(1, m18n.n('no_appslist_found'))
 
-    return { 'lists' : list_list }
+    # Migrate applist system if needed
+    # XXX move to a migration when those are implemented
+    if _using_legacy_applist_system():
+        _migrate_applist_system()
+
+    # Get the list
+    applist_list = _read_applist_list()
+
+    return applist_list
 
 
 def app_fetchlist(url=None, name=None):
     """
-    Fetch application list from app server
+    Fetch application list(s) from app server. By default, fetch all lists.
 
     Keyword argument:
-        name -- Name of the list (default yunohost)
-        url -- URL of remote JSON list (default https://app.yunohost.org/official.json)
-
+        name -- Name of the list
+        url -- URL of remote JSON list
     """
-    # Create app path if not exists
+    # If needed, create folder where actual applists are stored
     if not os.path.exists(REPO_PATH):
         os.makedirs(REPO_PATH)
 
-    if url is None:
-        url = 'https://app.yunohost.org/official.json'
-        name = 'yunohost'
-    elif name is None:
-        raise MoulinetteError(errno.EINVAL,
-                              m18n.n('custom_appslist_name_required'))
+    # Migrate applist system if needed
+    # XXX move that to a migration once they are finished
+    if _using_legacy_applist_system():
+        _migrate_applist_system()
+
+    # Read the list of applist...
+    applists = _read_applist_list()
```

Comment:
In fact I realized why I didn't do that in the first place (putting a call to `_migrate_applist_system()` in `_read_applist_list()`) : that's because `_migrate_applist_system()` calls `_register_new_applist` which calls `_read_applist_list()`, so that would create an infinite recursive call...